pkg/validation/spec/gnostic_test.go

```diff
@@ -294,158 +292,6 @@ func BenchmarkGnosticConversion(b *testing.B) {
 	}
 }
 
-// Ensure all variants of SecurityDefinition are being exercised by tests
-func TestSecurityDefinitionVariants(t *testing.T) {
-	type TestPattern struct {
-		Name    string
-		Pattern string
-	}
-
-	patterns := []TestPattern{
-		{
-			Name:    "Basic Authentication",
-			Pattern: `{"type": "basic", "description": "cool basic auth"}`,
-		},
-		{
-			Name:    "API Key Query",
-			Pattern: `{"type": "apiKey", "description": "cool api key auth", "in": "query", "name": "coolAuth"}`,
-		},
-		{
-			Name:    "API Key Header",
-			Pattern: `{"type": "apiKey", "description": "cool api key auth", "in": "header", "name": "coolAuth"}`,
-		},
-		{
-			Name:    "OAuth2 Implicit",
-			Pattern: `{"type": "oauth2", "flow": "implicit", "authorizationUrl": "https://google.com", "scopes": {"scope1": "a scope", "scope2": "a scope"}, "description": "cool oauth2 auth"}`,
-		},
-		{
-			Name:    "OAuth2 Password",
-			Pattern: `{"type": "oauth2", "flow": "password", "tokenUrl": "https://google.com", "scopes": {"scope1": "a scope", "scope2": "a scope"}, "description": "cool oauth2 auth"}`,
-		},
-		{
-			Name:    "OAuth2 Application",
-			Pattern: `{"type": "oauth2", "flow": "application", "tokenUrl": "https://google.com", "scopes": {"scope1": "a scope", "scope2": "a scope"}, "description": "cool oauth2 auth"}`,
-		},
-		{
-			Name:    "OAuth2 Access Code",
-			Pattern: `{"type": "oauth2", "flow": "accessCode", "authorizationUrl": "https://google.com", "tokenUrl": "https://google.com", "scopes": {"scope1": "a scope", "scope2": "a scope"}, "description": "cool oauth2 auth"}`,
-		},
-	}
-
-	for _, p := range patterns {
-		t.Run(p.Name, func(t *testing.T) {
-			// Parse JSON into yaml
-			var nodes yaml.Node
-			if err := yaml.Unmarshal([]byte(p.Pattern), &nodes); err != nil {
-				t.Error(err)
-				return
-			} else if len(nodes.Content) != 1 {
-				t.Errorf("unexpected yaml parse result")
-				return
-			}
-
-			root := nodes.Content[0]
-
-			parsed, err := openapi_v2.NewSecurityDefinitionsItem(root, compiler.NewContextWithExtensions("$root", root, nil, nil))
```

Comment:
NewSecurityDefinitionsItem's first parameter has to the the `yaml.Node` from `gopkg.in/yaml.v3` so we cannot use this method.